File: contest-management-client-server-java/ContestMgmtPersistence/src/main/java/contestmgmt/persistence/repository/ICrudRepository.java
package contestmgmt.persistence.repository;

import contestmgmt.model.Entity;

public interface ICrudRepository<ID, E extends Entity<ID>> {
    Iterable<E> findAll();
    E find(ID id);
    ID add(E entity);
    void remove(ID id);
    void update(E entity);
}

File: contest-management-cs/contest-management-gui-cs/repository/ICrudRepository.cs
using System.Collections.Generic;
using contest_management_gui_cs.model;

namespace contest_management_gui_cs.repository
{
    public interface ICrudRepository<ID, TE> where TE : Entity<ID>
    {
        IEnumerable<TE> FindAll();
        TE Find(ID id);
        void Add(TE entity);
        void Remove(ID id);
        void Update(TE entity);
    }
}